### PR TITLE
Loading indicators

### PR DESCRIPTION
**Pull request by christophscheuing**
see issue: #1082

Implemented mat-progress-bar as a loading indicator for the entity subrecord component and for the entity details component.

### CODE REVIEW

File: src/app/child-dev-project/notes/notes-manager/notes-manager.component.ts [hunk 1]
```diff
@@ -101,9 +102,12 @@ export class NotesManagerComponent implements OnInit {
 
   private async loadEntities(): Promise<Note[]> {
     let notes = await this.entityMapperService.loadType(Note);
+    this.isLoading = false;
     if (this.includeEventNotes) {
+      this.isLoading = true;
```
Comment (TheSlimvReal) [inline src/app/child-dev-project/notes/notes-manager/notes-manager.component.ts:107]: When is this needed?

File: src/app/core/entity-components/entity-list/entity-list.module.ts [hunk 2]
```diff
@@ -48,6 +49,7 @@ import { MatDialogModule } from "@angular/material/dialog";
     ExtendedModule,
     MatButtonToggleModule,
     MatTableModule,
+    MatProgressSpinnerModule,
```
Comment (TheSlimvReal) [inline src/app/core/entity-components/entity-list/entity-list.module.ts:52]:
```suggestion
```
This should not be needed here as the loading indicator is in the entity subrecord

File: src/app/core/entity/entity.module.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ import { EntityMapperService } from "./entity-mapper.service";
 import { EntitySchemaService } from "./schema/entity-schema.service";
 import { EntityConfigService } from "./entity-config.service";
 import { entityRegistry, EntityRegistry } from "./database-entity.decorator";
+import { MatProgressSpinnerModule } from "@angular/material/progress-spinner";
```
Comment (TheSlimvReal) [inline src/app/core/entity/entity.module.ts:24]:
```suggestion
```

**Comment by christophscheuing** [comment]
@TheSlimvReal: All your suggestions have been taken care of in the commit "Implemented spinner for add-day-attendance and acivity-list"

File: src/app/child-dev-project/notes/notes-manager/notes-manager.component.ts (hunk 1 above)
Comment (christophscheuing) [reply src/app/child-dev-project/notes/notes-manager/notes-manager.component.ts]: You're right, it's not needed.

File: src/app/core/entity-components/entity-details/entity-details.module.ts [hunk 4]
```diff
@@ -43,6 +45,8 @@ import { EntityUtilsModule } from "../entity-utils/entity-utils.module";
     MatMenuModule,
     FontAwesomeModule,
     EntityUtilsModule,
+    MatProgressSpinnerModule,
```
[the remark this answers is not in the document]
Comment (christophscheuing) [reply src/app/core/entity-components/entity-details/entity-details.module.ts]: Deleted this.

File: src/app/core/entity-components/entity-subrecord/entity-subrecord.module.ts [hunk 5]
```diff
@@ -58,6 +60,8 @@ import { MatMenuModule } from "@angular/material/menu";
     MatRippleModule,
     MatChipsModule,
     MatMenuModule,
+    MatProgressSpinnerModule,
```
[the remark this answers is not in the document]
Comment (christophscheuing) [reply src/app/core/entity-components/entity-subrecord/entity-subrecord.module.ts]: Right.

**Pull request merged**